src/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -285,15 +285,19 @@ internal async Task<NewCommandStatus> EnterUpdateFlowAsync(UpdateCommandArgs com
                     {
                         continue;
                     }
+                    if (!updatesToApply.Any(update => update.TemplatePackage is null || update.LatestVersion is null))
+                    {
+                        throw new Exception();
```

Comment:
Condition seems to be inverted here.

Plus please add some descriptive exception message